Sources/Penny/Handlers/MessageDeleteHandler.swift

```diff
@@ -41,29 +44,53 @@ struct MessageDeleteHandler {
     }
 }
 
-extension Payloads.CreateMessage {
+private extension Payloads.CreateMessage {
     init(
         messageCreate: Gateway.MessageCreate,
         author: DiscordUser
     ) {
-        let avatarURL = author.avatar.map {
-            CDNEndpoint.userAvatar(
-                userId: author.id,
-                avatar: $0
-            ).url
-        }
+        let member = messageCreate.member
+        let avatarURL = member?.uiAvatarURL ?? author.uiAvatarURL
+        let username = member?.uiName ?? author.username
+        let messageName = "message_\(messageCreate.id.rawValue)"
+        let jsonData = (try? JSONEncoder().encode(messageCreate)) ?? Data()
         self.init(
             embeds: [.init(
-                title: """
-                Message deleted from user \(DiscordUtils.mention(id: author.id)) with id: \(author.id.rawValue)
-                """,
-                description: messageCreate.content,
+                title: "Deleted Message",
+                description: DiscordUtils
+                    .escapingSpecialCharacters(messageCreate.content)
+                    .quotedMarkdown(),
                 timestamp: messageCreate.timestamp.date,
                 color: .red,
-                author: .init(
-                    name: author.username,
+                footer: .init(
+                    text: "From @\(username)",
                     icon_url: avatarURL.map { .exact($0) }
-                )
+                ),
+                fields: [
+                    .init(
+                        name: "Author",
+                        value: DiscordUtils.mention(id: author.id),
+                        inline: true
+                    ),
+                    .init(
+                        name: "Author ID",
+                        value: author.id.rawValue,
+                        inline: true
+                    ),
+                    .init(
+                        name: "Channel",
+                        value: DiscordUtils.mention(id: messageCreate.channel_id),
+                        inline: true
+                    )
+                ]
+            )],
+            files: [.init(
+                data: ByteBuffer(data: jsonData),
+                filename: messageName
+            )],
+            attachments: [.init(
+                index: 0,
+                filename: messageName
             )]
```

Comment:
include the whole message as an attachment for further investigations if needed.